lua/pixelui/core/cl_imgur.lua

```diff
@@ -43,9 +80,18 @@ function PIXEL.GetImgur(id, callback, useproxy, matSettings)
         function(error)
             if useproxy then
                 materials[id] = Material("nil")
-                return callback(materials[id])
+                return callback(materials[id], error)
             end
             return PIXEL.GetImgur(id, callback, true)
         end
     )
 end
+
+local downloadQueue = imgurQueue.new()
```

Comment:
There is no point making the imgur queue a class you can instantiate if there is only ever going to be one instance.